ui-tests/updateProfileValidation.spec.js

```diff
@@ -0,0 +1,275 @@
+import { test, expect } from "@playwright/test";
+import dotenv from "dotenv";
+import mongoose from "mongoose";
+import userModel from "../models/userModel";
+
+dotenv.config();
+
+test.beforeEach(async ({ page }) => {
+  // Connect to DB for creating and deleting test user
+  await mongoose.connect(process.env.MONGO_URL);
+
+  const hashedPassword =
+    "$2b$10$//wWsN./fEX1WiipH57HG.SAwgkYv1MRrPSkpXM38Dy5seOEhCoUy";
+
+  await new userModel({
+    name: "testupdate",
+    email: "testupdate@mail.com",
+    phone: "81234567",
+    address: "123 Test Road",
+    password: hashedPassword,
+    answer: "password is cs4218@test.com",
+  }).save();
+
+  // Navigate to HomePage
+  await page.goto("http://localhost:3000", { waitUntil: "commit" });
+  // Navigate to login page
+  await page.getByRole("link", { name: "Login" }).click();
+
+  // Log In as test user
+  await page.getByRole("textbox", { name: "Enter Your Email" }).click();
+  await page
+    .getByRole("textbox", { name: "Enter Your Email" })
+    .fill("testupdate@mail.com");
+  await page
+    .getByRole("textbox", { name: "Enter Your Password" })
+    .fill("cs4218@test.com");
+  await page.getByRole("button", { name: "LOGIN" }).click();
+});
+
+test.afterEach(async () => {
+  // Delete test user
+  await userModel.deleteMany({ email: "testupdate@mail.com" });
+
+  // Close connection to DB
+  await mongoose.disconnect();
+});
+
+test.describe("Update Profile Validation", () => {
+  test("should allow user to update profile successfully", async ({ page }) => {
```

Comment:
I already tested valid profile scenario in my test so just need to cover invalid scenario

Reply:
ok removed and added pairwise test case to replace